bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/bridge/AutomowerBridgeHandler.java

```diff
@@ -0,0 +1,162 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.automower.internal.bridge;
+
+import static org.openhab.binding.automower.internal.AutomowerBindingConstants.THING_TYPE_BRIDGE;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.automower.internal.rest.api.automowerconnect.dto.MowerListResult;
+import org.openhab.binding.automower.internal.rest.exceptions.AutomowerCommunicationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AutomowerBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Pfleger - Initial contribution
+ */
+@NonNullByDefault
+public class AutomowerBridgeHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(AutomowerBridgeHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+    private static final long DEFAULT_POLLING_INTERVAL_S = TimeUnit.HOURS.toSeconds(1);
+
+    private @Nullable ScheduledFuture<?> automowerBridgePollingJob;
+
+    private @Nullable AutomowerBridge bridge;
+    private final HttpClient httpClient;
+
+    private static class AutomowerBridgePollingRunnable implements Runnable {
+        private final Logger logger = LoggerFactory.getLogger(AutomowerBridgePollingRunnable.class);
+
+        private final AutomowerBridgeHandler handler;
+        private final AutomowerBridge bridge;
+
+        private AutomowerBridgePollingRunnable(AutomowerBridgeHandler handler, AutomowerBridge bridge) {
+            this.handler = handler;
+            this.bridge = bridge;
+        }
+
+        @Override
+        public void run() {
+            MowerListResult automowers;
+            try {
+                automowers = bridge.getAutomowers();
+                handler.updateStatus(ThingStatus.ONLINE);
+                logger.debug("Found {} automowers", automowers.getData().size());
+            } catch (AutomowerCommunicationException e) {
+                handler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error-query-mowers-failed");
+                logger.warn("Unable to fetch automowers: {}", e.getMessage());
+            }
+        }
+    }
```

Comment:
You can simple make this with a method in this class and call the method as `this::poller` in the `scheduleWithFixedDelay` call. (overlaps a bit with the other comment about making it a lambda)